src/KafkaFlow.Serializer/DefaultTypeResolver.cs

```diff
@@ -24,9 +28,11 @@ public void OnProduce(IMessageContext context)
 
             var messageType = context.Message.Value.GetType();
 
-            context.Headers.SetString(
-                MessageType,
-                $"{messageType.FullName}, {messageType.Assembly.GetName().Name}");
+            string messageTypeName = ProduceTypeCache.GetOrAdd(
+                messageType,
+                static messageType => $"{messageType.FullName}, {messageType.Assembly.GetName().Name}");
```

Comment:
I see benefits in using a Type cache on message consumption to avoid the Type.GetType call. But when producing I don't know if we will have a performance benefit changing from a string concatenation to a dictionary search.

Reply:
I think when producing there will be no performance benefit but rather a memory allocations benfit.